### ISSUE TITLE
can we find the Jupyter terminals which is running for long ?

### ISSUE THREAD

**Issue by mkaja** [jupyter/notebook#1834]
can we find the Jupyter terminals which is running for long or idle ? it takes system resources and leads to poor performance

**Comment by gnestor** [comment]
Are you aware of the Home > Running > Terminals tab? It lists all terminal instances. Are you proposing that terminal instances display _age_ like the notebook instances?

![image](https://cloud.githubusercontent.com/assets/512354/19405151/53e0069a-9229-11e6-9759-6e00d22404c4.png)

**Comment by mkaja** [comment]
yes Gnestor - Are you proposing that terminal instances display age like the notebook instances? yes that would be good . what we are trying to achieve is 100s of ppl  open the terminal and not using it for days the linux server is getting killed with high cpu and mem. if we can list the open terminals in linux command line -we can notify folks to close it properly.

**Comment by gnestor** [comment]
@Carreau @minrk I took a look through https://github.com/jupyter/notebook/blob/179bb24fbf79d153812858126127a91431da3319/notebook/static/tree/js/terminallist.js and it looks like a `created` property is not available on terminal items returned from the server. I also looked through https://github.com/jupyter/jupyter_server/blob/dd4b052403cc0ce70dbde9bfc6c71d6b09e814ec/notebook/terminal/api_handlers.py and it looks like terminado doesn't provide anything like a `created` property. Do you have an idea about how we could provide this info for terminal instances?

**Comment by minrk** [comment]
I'm not sure this information is present, but we can easily record a timestamp when a terminal is created. It might make more sense to follow #1827, though, and track activity on the websocket, so the stamp is last-activity, not created.

**Comment by gnestor** [comment]
@minrk Last activity is more similar to last modified (than created) which is what the "Running" tab in the dashboard exposes for notebooks. 

A couple questions regarding #1827: Does creating a terminal instance from the dashboard launch a kernel instance? If so, then would I just query the kernel model and use the `last_activity` value as the "last modified" value for a terminal instance?

**Comment by takluyver** [comment]
No, terminals do not use kernels.

**Comment by tuliocasagrande** [comment]
I added `/api/terminals/` to the documentation: https://github.com/jupyter/jupyter/wiki/Jupyter-Notebook-Server-API#terminals-api, but it doesn't return the state indeed.
